hal_st/synchronous_stm32fxxx/SynchronousAesStm.cpp

```diff
@@ -0,0 +1,133 @@
+#include "hal_st/synchronous_stm32fxxx/SynchronousAesStm.hpp"
+#include "infra/util/MemoryRange.hpp"
+#include "infra/util/ReallyAssert.hpp"
+
+#if defined(HAS_PERIPHERAL_AES)
+
+namespace hal
+{
+    SynchronousAesEcbStm::SynchronousAesEcbStm(const Config& config)
+        : config(config)
+    {}
+
+    SynchronousAesEcbStm::~SynchronousAesEcbStm()
+    {
+        Disable();
+    }
+
+    void SynchronousAesEcbStm::SetKey(const std::array<uint8_t, 16>& key) const
+    {
+        std::copy_n(key.begin(), key.size(), localKey.begin());
+    }
+
+    void SynchronousAesEcbStm::Encrypt(infra::ConstByteRange input, infra::ByteRange output) const
+    {
+        Enable();
+        SetKey();
```

Comment:
Although SetKey() only modifies 4 registers, setting up an AES key takes considerable calculations before encryption can start. My suspicion is therefore that setting the key each encryption cycle delays the whole encryption by a considerable amount. Please consider moving SetKey() to the SetKey(...) member function